pysatMadrigal/instruments/gnss_tec.py

```diff
@@ -0,0 +1,216 @@
+# -*- coding: utf-8 -*-.
+"""Supports the MIT Haystack GNSS TEC data products
+
+The Global Navigation Satellite System (GNSS) is used in conjunction with a
+world-winde receiver network to produce total electron content (TEC) data
+products, including vertical and line-of-sight TEC.
+
+Downloads data from the MIT Haystack Madrigal Database.
+
+Properties
+----------
+platform
+    'gnss'
+name
+    'tec'
+tag
+    'vtec', 'los'
+
+Examples
+--------
+::
+
+    import datetime
+    import pysat
+    import pysatMadrigal as pymad
+
+    vtec = pysat.Instrument(inst_module=pymad.instruments.gnss_tec, tag='vtec')
+    vtec.download(dt.datetime(2017, 11, 19), dt.datetime(2017, 11, 20),
+                  user='Firstname+Lastname', password='email@address.com')
+    vtec.load(date=dt.datetime(2017, 11, 19))
+
+
+Note
+----
+    Please provide name and email when downloading data with this routine.
+
+"""
+
+import datetime as dt
+import functools
+import numpy as np
+
+from pysat.instruments.methods import general as ps_gen
+
+from pysatMadrigal.instruments.methods import madrigal as mad_meth
+
+import logging
+logger = logging.getLogger(__name__)
+
+
+platform = 'gnss'
+name = 'tec'
+tags = {'vtec': 'vertical TEC', 'los': 'slant TEC'}
+sat_ids = {'': list(tags.keys())}
+_test_dates = {'': {'vtec': dt.datetime(2017, 11, 19),
+                    'los': dt.datetime(2017, 11, 19)}}
+pandas_format = False
+
+# support list files routine
+# use the default pysat method
+dname = '{year:4d}{month:02d}{day:02d}'
```

Comment:
I'm able to successfully download the files, but they use a two-digit year
```
gps171119g.002.hdf5
```

Reply:
There is also an issue in `ps_gen.list_files`. The two_digit_year_break keyword is not passed to the underlying from_os constructor. I have a working version locally.

`dname = '{year:2d}{month:02d}{day:02d}'`

Reply:
Did not notice that.